src/datasources/transaction-api/transaction-api.service.ts

```diff
@@ -199,7 +199,7 @@ export class TransactionApi implements ITransactionApi {
       JSON.stringify(isSafe),
       isSafe
         ? // We can indefinitely cache this as an address cannot "un-Safe" itself
-          TransactionApi.INDEFINITE_EXPIRATION_TIME
+          TransactionApi.MAX_TTL
```

Comment:
Considering `-1` is a "valid" TTL value, do you think we should intead modify [the client](https://github.com/safe-global/safe-client-gateway/blob/5a66c242d50fc32868b0bfb5c8a4467163053801/src/datasources/cache/redis.cache.service.ts#L39) to allow it instead? We otherwise lose the ability to cache indefinitely.

Should we want to prevent negative numbers entirely, we should remove the above comment.

Reply:
I agree, I thought about the same thing while fixing this, but I'm pretty confident that modifying the client would have an impact in other places of the code because if I recall correctly `CacheFirstDataSource.get` is called without an `expireTimeSeconds` value on purpose to prevent `CacheService` from caching the result.

So while I agree with your suggestion, I'd like to do it in a separate PR if you agree.

Reply:
I found an example: https://github.com/safe-global/safe-client-gateway/blob/5a66c242d50fc32868b0bfb5c8a4467163053801/src/datasources/transaction-api/transaction-api.service.ts#L251

Reply:
Let's keep it as is then - the TTL equates to just over 68 years!

Reply:
I've set it to `Number.MAX_SAFE_INTEGER - 1` in [bd424ec](https://github.com/safe-global/safe-client-gateway/pull/1757/commits/bd424ec02276ec00e9092172e52773772ca832d6) and I've added a note elaborating this decision. Thanks for the input!